gr-zeromq/include/gnuradio/zeromq/pub_sink.h

```diff
@@ -57,6 +57,11 @@ namespace gr {
        */
       static sptr make(size_t itemsize, size_t vlen, char *address,
                        int timeout=100, bool pass_tags=false, int hwm=-1);
+
+      /*!
+       * \brief If the endpoint is TCP return the port number; -1 otherwise.
+       */
+      virtual int port() = 0;
```

Comment:
Instead of adding this to all the variations, the best would be to have a base.h for all the ZMQ sink/source and put the common methods in there.